cpp/velox/memory/VeloxMemoryManager.cc

```diff
@@ -15,91 +15,15 @@
  * limitations under the License.
  */
 
-#include "VeloxMemoryPool.h"
-#include "compute/Backend.h"
-#include "compute/VeloxBackend.h"
-#include "compute/VeloxInitializer.h"
-#include "utils/TaskContext.h"
-#include "velox/common/memory/MemoryAllocator.h"
+#include "VeloxMemoryManager.h"
 
-#include <glog/logging.h>
-
-using namespace facebook;
+#include "utils/exception.h"
 
 namespace gluten {
 
-// So far HbmMemoryAllocator would not work correctly since the underlying
-//   gluten allocator is only used to do allocation-reporting to Spark in mmap case
-class VeloxMemoryAllocator final : public velox::memory::MemoryAllocator {
- public:
-  VeloxMemoryAllocator(gluten::MemoryAllocator* glutenAlloc, velox::memory::MemoryAllocator* veloxAlloc)
-      : glutenAlloc_(glutenAlloc), veloxAlloc_(veloxAlloc) {}
-
-  Kind kind() const override {
-    return veloxAlloc_->kind();
-  }
-
-  bool allocateNonContiguous(
-      velox::memory::MachinePageCount numPages,
-      velox::memory::Allocation& out,
-      ReservationCallback reservationCB,
-      velox::memory::MachinePageCount minSizeClass) override {
-    return veloxAlloc_->allocateNonContiguous(numPages, out, reservationCB, minSizeClass);
-  }
-
-  int64_t freeNonContiguous(velox::memory::Allocation& allocation) override {
-    int64_t freedBytes = veloxAlloc_->freeNonContiguous(allocation);
-    return freedBytes;
-  }
-
-  bool allocateContiguous(
-      velox::memory::MachinePageCount numPages,
-      velox::memory::Allocation* collateral,
-      velox::memory::ContiguousAllocation& allocation,
-      ReservationCallback reservationCB) override {
-    return veloxAlloc_->allocateContiguous(numPages, collateral, allocation, reservationCB);
-  }
-
-  void freeContiguous(velox::memory::ContiguousAllocation& allocation) override {
-    veloxAlloc_->freeContiguous(allocation);
-  }
-
-  void* allocateBytes(uint64_t bytes, uint16_t alignment) override {
-    void* out;
-    VELOX_CHECK(glutenAlloc_->allocateAligned(alignment, bytes, &out))
-    return out;
-  }
-
-  void freeBytes(void* p, uint64_t size) noexcept override {
-    VELOX_CHECK(glutenAlloc_->free(p, size));
-  }
-
-  bool checkConsistency() const override {
-    return veloxAlloc_->checkConsistency();
-  }
-
-  velox::memory::MachinePageCount numAllocated() const override {
-    return veloxAlloc_->numAllocated();
-  }
-
-  velox::memory::MachinePageCount numMapped() const override {
-    return veloxAlloc_->numMapped();
-  }
-
-  std::string toString() const override {
-    return veloxAlloc_->toString();
-  }
-
-  size_t capacity() const override {
-    return veloxAlloc_->capacity();
-  }
-
- private:
-  gluten::MemoryAllocator* glutenAlloc_;
-  velox::memory::MemoryAllocator* veloxAlloc_;
-};
```

Comment:
Any particular reason to remove this code?

Reply:
Only delegate two API is dangerous, and before this patch, glutenAlloc is bind with listener as ListenableAllocator, this would cause extra fake bytes usage.

We could implement custom allocator in NativeMemoryAllocator with new type and passed to NativeMemoryManager when needed.